Controller/Admin/Transaction/TransactionTabPostProcessing.php

```diff
@@ -157,68 +294,140 @@ private function _processRequest(array $aRequestParameters)
     /**
      * Returns an array of processing actions to be displayed below the amount input field.
      *
+     * @throws Exception in case the payment method type is not found
+     *
      * @return array
      *
-     * @since 1.0.0
+     * @since 1.0.1
      */
-    protected function _getPostProcessingActions(): array
+    protected function _getPostProcessingActions()
     {
-        // TODO return only supported actions for the payment
-        return [
-            'cancel' => [
-                'title' => Helper::translate('wd_cancel'),
-                'callback' => '_onCancel',
-            ],
-            'capture' => [
-                'title' => Helper::translate('wd_pay'),
-                'callback' => '_onCapture',
-            ],
-            'refund' => [
-                'title' => Helper::translate('wd_refund'),
-                'callback' => '_onRefund',
-            ],
+        $sPaymentId = $this->oTransaction->getPaymentType();
+
+        // Need to create a transaction object with the ID of the currently selected one to get
+        // the available post-processing operations from the Payment SDK
+        $oPaymentMethod = PaymentMethodFactory::create($sPaymentId);
+        $oTransaction = $oPaymentMethod->getTransaction();
+        $sParentTransactionId = $this->oTransaction->wdoxidee_ordertransactions__transactionid->value;
+        $oTransaction->setParentTransactionId($sParentTransactionId);
+
+        $oConfig = $this->_getPaymentMethodConfig();
+        $aPossibleOperations = $this->_getBackendService($oConfig)->retrieveBackendOperations($oTransaction, true);
+
+        if ($aPossibleOperations === false || count($aPossibleOperations) <= 0) {
+            return [];
+        }
+
+        $aPossibleOperations = $this->_filterPostProcessingActions($aPossibleOperations, $oPaymentMethod);
+
+        return $this->_getTranslatedPostProcessingActions($aPossibleOperations);
+    }
+
+    /**
+     * Filters the returned post processing actions for a payment method.
+     * It is possible to do payment method specific modifications in this method.
+     *
+     * @param array         $aPossibleOperations
+     * @param PaymentMethod $oPaymentMethod
+     *
+     * @return array
+     *
+     * @since 1.0.1
+     */
+    private function _filterPostProcessingActions($aPossibleOperations, $oPaymentMethod)
+    {
+        return array_filter($aPossibleOperations, function () use ($oPaymentMethod) {
+            // currently there are no post-processing transactions for Sofort
+            return $oPaymentMethod->getName() !== SofortTransaction::NAME;
+        }, ARRAY_FILTER_USE_KEY);
```

Comment:
`ARRAY_FILTER_USE_KEY` is unnecessary, the key is not used anywhere.

Reply:
Thanks, actually removed the whole `array_filter` now as it is currently useless.